src/DynamoCore/Models/DynamoModelCommands.cs

```diff
@@ -362,21 +362,31 @@ private void BeginShiftReconnections(Guid nodeId, int portIndex, PortType portTy
             PortModel selectedPort = node.OutPorts[portIndex];
 
             var connectorsForDeletion = new List<ModelBase>();
-            int numOfConnectors = selectedPort.Connectors.Count;
+
+            var selectedConnectors = new List<ConnectorModel>();
+            selectedConnectors = selectedPort.Connectors.Where(x => x.End.Owner.IsSelected).ToList();
+
+            // If no connectors are selected, process all of the associated nodes
+            if (selectedConnectors.Count() <= 0)
+            {
+                selectedConnectors = selectedPort.Connectors.ToList();
+            }
+
+            int numOfConnectors = selectedConnectors.Count();
             if (numOfConnectors == 0) return;
             
             activeStartPorts = new PortModel[numOfConnectors];
 
             for (int i = 0; i < numOfConnectors; i++)
             {
-                ConnectorModel connector = selectedPort.Connectors[i];
+                ConnectorModel connector = selectedConnectors[i];
                 connectorsForDeletion.Add(connector);
```

Comment:
`connectorsForDeletion` seems a dup to me after the refactor? What do you think

Reply:
I think it's still required as removing it will break undo/redo.  Below we call `CurrentWorkspace.SaveAndDeleteModels(connectorsForDeletion)` where `SaveAndDeleteModels()` is a method in `UndoRedo.cs` which is a partial class of the `WorkspaceModel`.  Several new and old recorded tests break as well.

Reply:
I understand before the refactor it is required, but after the refactor, it contains exactly all the connectors  in `selectedConnectors`.
Can we call code below to preserve the undo/redo behavior?
```
CurrentWorkspace.SaveAndDeleteModels(selectedConnectors);
```